spring-integration-core/src/main/kotlin/org/springframework/integration/dsl/KotlinIntegrationFlowDefinition.kt

```diff
@@ -971,31 +977,33 @@ class KotlinIntegrationFlowDefinition(@PublishedApi internal val delegate: Integ
 
 	/**
 	 * Populate a [ScatterGatherHandler] to the current integration flow position
-	 * based on the provided [RecipientListRouterSpec] for scattering function
+	 * based on the provided [KotlinRecipientListRouterSpec] for scattering function
 	 * and default [AggregatorSpec] for gathering function.
 	 */
-	fun scatterGather(scatterer: RecipientListRouterSpec.() -> Unit) {
-		this.delegate.scatterGather(scatterer)
+	fun scatterGather(scatterer: KotlinRecipientListRouterSpec.() -> Unit) {
+		this.delegate.scatterGather(Consumer { scatterer(KotlinRecipientListRouterSpec(it)) })
 	}
 
 	/**
-	 * Populate a [ScatterGatherHandler] to the current integration flow position
-	 * based on the provided [RecipientListRouterSpec] for scattering function
+	 * Populate a [KotlinRecipientListRouterSpec] to the current integration flow position
```

Comment:
This looks suspicious. The function is `scatterGather` and the variant below does not have this change.

Reply:
Doh...

Will fix soon